File: Algorithm/BOJ/C++/10872.cpp
#include <iostream>

using namespace std;



class Factorial
{
public:
	int getFactorial(int n);
};



int Factorial::getFactorial(int n) {
	
	if ( n >= 2 ) {
		return n * getFactorial(n - 1);
	}
	else {
		return 1;
	}
	
}


int main() {

	int n;
	cin >> n;

	Factorial factorial;
	cout << factorial.getFactorial(n);
}

File: Algorithm/BOJ/Go/10872.go
// 10872
package main

import (
	"fmt"
)

func getFactorial(n int) int {
	if n >= 2 {
		return n * getFactorial(n-1)
	} else {
		return 1
	}
}

func main() {
	var n int
	fmt.Scanln(&n)

	fmt.Println(getFactorial(n))
}
